homeassistant/components/zha/switch.py

```diff
@@ -138,3 +150,120 @@ async def async_update(self) -> None:
 
         self._state = len(on_states) > 0
         self._available = any(state.state != STATE_UNAVAILABLE for state in states)
+
+
+class ZHASwitchConfigurationEntity(ZhaEntity, SwitchEntity):
+    """Representation of a ZHA switch configuration entity."""
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _zcl_attribute: str
+    _inverted = False
+
+    @classmethod
+    def create_entity(
+        cls,
+        unique_id: str,
+        zha_device: ZHADevice,
+        channels: list[ZigbeeChannel],
+        **kwargs,
+    ) -> ZhaEntity | None:
+        """Entity Factory.
+
+        Return entity if it is a supported configuration, otherwise return None
+        """
+        channel = channels[0]
+        if (
+            cls._zcl_attribute in channel.cluster.unsupported_attributes
+            or channel.cluster.get(cls._zcl_attribute) is None
+        ):
+            _LOGGER.debug(
+                "%s is not supported - skipping %s entity creation",
+                cls._zcl_attribute,
+                cls.__name__,
+            )
+            return None
+
+        return cls(unique_id, zha_device, channels, **kwargs)
+
+    def __init__(
+        self,
+        unique_id: str,
+        zha_device: ZHADevice,
+        channels: list[ZigbeeChannel],
+        **kwargs,
+    ) -> None:
+        """Init this number configuration entity."""
+        self._channel: ZigbeeChannel = channels[0]
+        super().__init__(unique_id, zha_device, channels, **kwargs)
+
+    async def async_added_to_hass(self) -> None:
+        """Run when about to be added to hass."""
+        await super().async_added_to_hass()
+        self.async_accept_signal(
+            self._channel, SIGNAL_ATTR_UPDATED, self.async_set_state
+        )
+
+    @callback
+    def async_set_state(self, attr_id: int, attr_name: str, value: Any):
+        """Handle state update from channel."""
+        self.async_write_ha_state()
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the switch is on based on the statemachine."""
+        val = self._channel.cluster.get(self._zcl_attribute)
+        if val is None:
+            return False if not self._inverted else True
+        return bool(val) if not self._inverted else (not bool(val))
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn the entity on."""
+        try:
+            result = await self._channel.cluster.write_attributes(
+                {self._zcl_attribute: 1 if not self._inverted else 0}
```

Comment:
I'm currently using 1 or 0 in the attribute what I want to switch. Will be better to switch to boolean?